rejectedsoftware/vibe.d
# Add reading encoded file from disk if present

This commit adds the ability to send compressed files if present
instead of compressing the file on each request.

To test it out

```
$ cd examples/http_static_server/
$ dub --build=release
$ ab -H "Accept-Encoding: gzip, deflate" -r -n 10000 -c 100 http://localhost:8080/index_large.html
[snip]
$ ab -H "Accept-Encoding: gzip, deflate" -r -n 10000 -c 100 http://localhost:8080/gzip/index_large.html
[snip]
```

This fixes part of #143. I'd like to see the needed change for libevent. I'm curious what's the impact on performance. Is somebody working on it?
BTW do you have some performance numbers for vibe.d handling static files. I only get about 10MB per second. Must be doing something wrong.
